### PR TITLE
image: Fix options.imagebitmap on Firefox

### PR DESCRIPTION
**Pull request by ibgreen**
As reported by @isaacbrodsky, Firefox crashes if options are passed to `createImageBitmap`.

This fix chooses to still support `options.imagebitmap` on browsers that do support that parameter. 

@Pessimistress While trying to test this fix, I realized that the page served by `yarn test browser` is dependent on puppeteer and can't be run in other browsers. Would be nice to have the option to easily run our unit tests in other browsers. Not sure if we could just make the puppeteer calls optional if not run on Chromium, or if we'd need a separate ocular sub target.

### CODE REVIEW

File: modules/images/src/lib/parsers/parse-to-image-bitmap.js [hunk 1]
```diff
@@ -4,9 +4,33 @@
...
+    } catch (error) {
+      console.warn(error); // eslint-disable-line
+      imagebitmapOptionsSupported = false;
+    }
+  }
+  return createImageBitmap(blob);
+}
+
+function isEmptyObject(object) {
+  for (const key in object) {
```
Comment (Pessimistress) [inline modules/images/src/lib/parsers/parse-to-image-bitmap.js]: This doesn’t seem to account for object===undefined

**Comment by Pessimistress** [comment]
> yarn test browser is dependent on puppeteer and can't be run in other browsers

Ye we can make the dependency optional. Render tests won’t work because the screen capture and comparison is done on the server side.

Puppeteer has experimental support for Firefox but I have never tried.

File: modules/images/src/lib/parsers/parse-to-image-bitmap.js (hunk 1 above)
Comment (ibgreen) [reply modules/images/src/lib/parsers/parse-to-image-bitmap.js]: > This doesn’t seem to account for object===undefined

Shouldn't happen due to the way options are merged with defaults that said adding the check doesn't cost much., will do.

**Comment by ibgreen** [comment]
> Puppeteer has experimental support for Firefox but I have never tried.

That's cool, it would be great if we could have dual browser test automation, though I'd prioritize the puppeteer-less mode so that we can run unit tests in Safari and other browsers too.

File: modules/images/src/lib/parsers/parse-to-image-bitmap.js (hunk 1 above)
Comment (Pessimistress) [reply modules/images/src/lib/parsers/parse-to-image-bitmap.js]: Looking at the previous attempt: https://github.com/uber-web/loaders.gl/commit/d348845a489739a1464e5e27ac6c5e8764cdd041

Does it throw immediately or does the promise reject?

Comment (ibgreen) [reply modules/images/src/lib/parsers/parse-to-image-bitmap.js]: > Does it throw immediately or does the promise reject?

Yes I believe you have a point. `async await` converts automatically between promise rejections and exceptions, so I'll update to use async syntax.

Still this makes it obvious that I have not been able to test these changes, due to our test harness limitations.

Comment (Pessimistress) [reply modules/images/src/lib/parsers/parse-to-image-bitmap.js]: Could you hold on merging while I fix the testing?

**Comment by Pessimistress** [comment]
> I have not been able to test these changes, due to our test harness limitations.

https://github.com/uber-web/probe.gl/pull/124

**Comment by ibgreen** [comment]
Image tests are passing on Firefox (however the new setup uncovered two stream related errors which is great). Will follow up on those separately.

**Pull request merged**